### PR TITLE
Change bitfields into EnumSets #381

### PR DESCRIPTION
**Pull request by DeepSnowNeeL**
Changed every int modifiers to an EnumSet<Modifier>, added enum Modifier and fixed tests

### CODE REVIEW

**Comment by DeepSnowNeeL** [comment]
Allright all tests pass now :)

**Comment by matozoid** [comment]
Check out #57 and see if there's something useful there please

**Comment by DeepSnowNeeL** [comment]
I already done something for #57 in my other current work (builder methods & helpers)
(See https://github.com/DeepSnowNeeL/javaparser/commit/9eae286ac348dbab1998a8c79901d190436f474d , I'll just need to refactor )

You should wait for my other work with builder methods to accept this PR, so we can release all this in the next major release :)

**Comment by matozoid** [comment]
My thought was to start work on 3.0.0 after 2.5.0. I'll have so much fun deleting deprecated stuff! :) :) :)

**Comment by matozoid** [comment]
Do you think we can get rid of ModifierSet?

File: javaparser-core/src/main/java/com/github/javaparser/ast/body/FieldDeclaration.java [hunk 1]
```diff
@@ -31,18 +38,12 @@
 import com.github.javaparser.ast.visitor.GenericVisitor;
 import com.github.javaparser.ast.visitor.VoidVisitor;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import static com.github.javaparser.Position.pos;
-import static com.github.javaparser.ast.internal.Utils.*;
-
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/body/FieldDeclaration.java:39]: Be careful with automatically organizing imports, it causes a lot of nonsense in the commits.

**Comment by DeepSnowNeeL** [comment]
Yes if you want to ?
It will just break more code.. ?

Le lun. 18 juil. 2016 21:18, Danny van Bruggen notifications@github.com a
écrit :

**Comment by matozoid** [comment]
Pretty good stuff for the rest :)

**Comment by matozoid** [comment]
We're going for 3.0, so let's clean up as much as possible!

... we have to start thinking of a migration guide already...

**Comment by DeepSnowNeeL** [comment]
Allright I will do it.
About the imports, it imports only whats used ? Is it so bad ?
I agree tho, the commit diff looks stupid because of this :p

Le lun. 18 juil. 2016 21:23, Danny van Bruggen notifications@github.com a
écrit :

**Comment by matozoid** [comment]
Your imports are fine, there was just no need to change them. It might be that my IDE settings are different, and when I do "organize imports" on the same files, all the imports will change again. All possible merge conflicts and long, long pull requests for very little benefit.

**Comment by matozoid** [comment]
I'm waiting for your stuff now, because you seem to be changing so much that I don't want to get into conflicts with it.

**Comment by DeepSnowNeeL** [comment]
I wish we could configure it so it imports the same way, I'll disable auto
import for the next commits (but tbh, I already refactored almost all
classes in my other branch :p)

Le lun. 18 juil. 2016 21:27, Danny van Bruggen notifications@github.com a
écrit :

**Comment by DeepSnowNeeL** [comment]
Allright, I could use some help about tests for the builder branch, I'll pm
you on gitter

Le lun. 18 juil. 2016 21:29, Maximilien Cruz maximilien.cruz@gmail.com a
écrit :

**Comment by matozoid** [comment]
So, is this PR okay? The code seems fine.

File: javaparser-core/src/main/java/com/github/javaparser/ast/body/Modifier.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	PROTECTED("protected"),
+	STATIC("static"),
+	FINAL("final"),
+	SYNCHRONIZED("synchronized"),
+	VOLATILE("volatile"),
+	TRANSIENT("transient"),
+	NATIVE("native"),
+	ABSTRACT("abstract"),
+	STRICTFP("strictfp");
+
```
Comment (SmiddyPence) [inline javaparser-core/src/main/java/com/github/javaparser/ast/body/Modifier.java:19]: Maybe go with the JLS suggested order, http://checkstyle.sourceforge.net/config_modifier.html

Doesn't technically matter, be reinforces the principal.

```
public
protected
private
abstract
static
final
transient
volatile
synchronized
native
strictfp
```

**Comment by SmiddyPence** [comment]
👍  nice improvement

**Comment by DeepSnowNeeL** [comment]
Let's close this one since it was included in #400 :)

**Comment by matozoid** [comment]
Can you submit a PR for that thing that @SmiddyPence said?

**Pull request closed without merging**